test/functionTests-TellorFlex.js

```diff
@@ -881,37 +908,54 @@ describe("TellorFlex - Function Tests", function () {
 		expect(await tellor.accumulatedRewardPerShare()).to.equal(expAccumRewPerShare)
 
 		// deposit another stake
-		await token.mint(accounts[1].address, h.toWei("100"))
-		// await tellor.connect(accounts[1]).depositStake(h.toWei("50"))
-		await tellor.updateRewards()
+		expect(await token.balanceOf(accounts[1].address)).to.equal(h.toWei("900"))
+		await tellor.connect(accounts[1]).depositStake(h.toWei("50"))
 		blocky3 = await h.getBlock()
 
-		expect(await tellor.timeOfLastAllocation()).to.equal(blocky3.timestamp)
+		// state checks
+		expAccumRewPerShare = expAccumRewPerShare + (BigInt(blocky3.timestamp - blocky2.timestamp)) * BigInt(expectedRewardRate) * BigInt(1e18) / BigInt(100e18)
+		expect(await tellor.accumulatedRewardPerShare()).to.equal(expAccumRewPerShare)
+		expectedPayout = expAccumRewPerShare * BigInt(100e18) / BigInt(1e18)
+		expectedBal = BigInt(h.toWei("900")) - BigInt(h.toWei("50")) + expectedPayout
+		expect(await token.balanceOf(accounts[1].address)).to.equal(expectedBal)
+		expTotalRewardDebt = BigInt(150e18) * expAccumRewPerShare / BigInt(1e18)
+		expect(await tellor.totalRewardDebt()).to.equal(expTotalRewardDebt)
+		expStakingRewardsBal = BigInt(STAKING_REWARDS1) - expectedPayout
+		expect(await tellor.stakingRewardsBalance()).to.equal(expStakingRewardsBal)
+
+		// update rewards
+		await tellor.updateRewards()
+		blocky4 = await h.getBlock()
+
+		// state checks
+		expect(await tellor.timeOfLastAllocation()).to.equal(blocky4.timestamp)
 		expect(await tellor.rewardRate()).to.equal(expectedRewardRate)
-		expAccumRewPerShare = expAccumRewPerShare + (BigInt(blocky3.timestamp - blocky2.timestamp) * BigInt(expectedRewardRate) * BigInt(1e18) / BigInt(100e18))
+		expAccumRewPerShare = expAccumRewPerShare + (BigInt(blocky4.timestamp - blocky3.timestamp) * BigInt(expectedRewardRate) * BigInt(1e18) / BigInt(150e18))
 		expect(await tellor.accumulatedRewardPerShare()).to.equal(expAccumRewPerShare)
-		expectedStakingRewardsBal = BigInt(STAKING_REWARDS1) - (expAccumRewPerShare * BigInt(h.toWei("100")))
```

Comment:
Does the state of the staking rewards balance need to be checked again after the call to `updateRewards` on line 927?